### DISCUSSION TITLE
Native types mapping

### DISCUSSION THREAD

**Discussion by williamolojede** [prisma/prisma#7237] [Q&A] [answered]
I just read in the docs that Prisma supports native types mapping using the @db.XXX syntax since v2.17.0, I've upgraded to the latest version of both the cli and client package but when I run `prisma migrate` i get an error `Error validating: This line is not a valid field or attribute definition.`. Is there an extra config I need to do?

**Comment by ryands17** [comment, marked as answer]
@williamolojede 👋 

Could you share your schema? Also Native types do not work with SQLite.

**Reply by williamolojede** [reply]
Hi @ryands17, I use a Postgres DB

```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["createMany"]
}

model Offer {
  id              String    @id @default(uuid())
  amount            @db.Decimal(9, 3)
  createdAt DateTime @default(now()) @map(name: "created_at")
  updatedAt DateTime @updatedAt @map(name: "updated_at")

  @@map(name: "offer")
}
```

**Reply by ryands17** [reply]
You're missing the `Decimal` before `@db.Decimal`. It should be:

```prisma
model Offer {
  id        String   @id @default(uuid())
  amount    Decimal  @db.Decimal(9, 3)
  createdAt DateTime @default(now()) @map(name: "created_at")
  updatedAt DateTime @updatedAt @map(name: "updated_at")

  @@map(name: "offer")
}
```

**Reply by williamolojede** [reply]
OMG, thanks, Ryan.

**Reply by anon-ghub** [reply]
The docs need to provide more examples with the full line.
